source/patterns/@aws-solutions-constructs/aws-lambda-kendra/lib/index.ts

```diff
@@ -0,0 +1,197 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+// Imports
+import * as defaults from "@aws-solutions-constructs/core";
+import * as lambda from "aws-cdk-lib/aws-lambda";
+import * as iam from "aws-cdk-lib/aws-iam";
+import * as kendra from "aws-cdk-lib/aws-kendra";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import { Construct } from "constructs";
+
+/**
+ * @summary The properties for the LambdaToKendra class.
+ */
+export interface LambdaToKendraProps {
+  /**
+   *
+   *
+   * @default - Optional user provided props to override the default props for the Kendra index. Is this required?
+   */
+  readonly kendraIndexProps?: kendra.CfnIndexProps;
+  /**
+   * A list of data sources that will provide data to the Kendra index. ?At least 1 must be specified. We will do majority of
+   * processing for some data sources (S3 crawler initially), but for others the props must be complete (e.g. proper roleArn, etc.)
+   *
+   * @default - empty list (no data sources)
+   */
+  readonly kendraDataSourcesProps: Array<kendra.CfnDataSourceProps | any>;
+  /**
+   * Optional - index permissions to grant to the Lambda function. One or more of the following
+   * may be specified: `Read`, `SubmitFeedback` and `Write`. Default is `["Read", "SubmitFeedback"]`. Read is
+   * all the operations IAM defines as Read and List. SubmitFeedback is only the SubmitFeedback action. Write is all the
+   * operations IAM defines as Write and Tag. This functionality may be overridden by providing a specific role arn in lambdaFunctionProps
+   *
+   * @default - ["Read", "SubmitFeedback"]
```

Comment:
Do these string values come from the aws-cdk-lib or from us?
I was going to leave a comment here about putting the three string values for index permissions (read, submitfeedback, write) into an enum, but then backtracked. I'll leave it here as a consideration.